lib/util/pool.ts

```diff
@@ -0,0 +1,34 @@
+import { configurationValue } from "../configuration";
+
+function concurrentDefault(): number {
+    return configurationValue<number>("pool.concurrent", 5);
+}
+
+/**
+ * Execute all provided promises with a max concurrency
+ * @param promises all promises to execute
+ * @param concurrent the max number of concurrent promise executions
+ */
+export async function executeAll<T>(promises: Array<Promise<T>>, concurrent: number = concurrentDefault()): Promise<T[]> {
```

Comment:
I did not know you could use a call a function to provide a default value. I guess because that is all actually managed at run time, not compile time.